components/camel-elsql/src/main/java/org/apache/camel/component/elsql/ElsqlEndpoint.java

```diff
@@ -121,16 +123,39 @@ protected void doStart() throws Exception {
             elSqlConfig = ElSqlDatabaseVendor.Default.asElSqlConfig();
         }
 
+        // load and parse the sources which are from classpath
+        parseResources(resourceUri, uri-> ResourceHelper.isClasspathUri(uri));
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        // load and parse the sources which are not from classpath
+        parseResources(resourceUri, uri-> !ResourceHelper.isClasspathUri(uri));
+    }
+
+    private void parseResources(String resourceUri, Predicate<String> predicate) throws Exception {
         // there can be multiple resources
         // so we have all this lovely code to turn that into an URL[]
         final List<URL> list = new ArrayList<>();
         final Iterable it = ObjectHelper.createIterable(resourceUri);
-        for (final Object path : it) {
-            final URL url = ResourceHelper.resolveMandatoryResourceAsUrl(getCamelContext().getClassResolver(), path.toString());
-            list.add(url);
+
+        StreamSupport.stream(it.spliterator(), false)
+                .filter(predicate)
+                .forEach(path -> {
+                    try {
+                        final URL url = ResourceHelper.resolveMandatoryResourceAsUrl(getCamelContext().getClassResolver(), path.toString());
+                        list.add(url);
+                    } catch (Exception e) {
+
```

Comment:
This is wrong as you ignore thrown exception - I suggest to NOT use streaming API as its also TERRIBLE sometimes as in this example being overly complex and causing you to do try .. catch and ignore exceptions.

Reply:
OK, I see and will update to use the iterable as usual.

Reply:
Thanks